source/fundamentals/data-formats/document-data-format-record.txt

```diff
@@ -225,4 +230,20 @@ execute a find operation on a collection of ``RecordTree`` types:
    .. output::
       :language: sh
 
-      RecordTree[content=Ranginui, left=RecordTree[content=..., left=RecordTree[content=..., right=RecordTree[content=Ikatere...]]
\ No newline at end of file
+      RecordTree[content=Ranginui, left=RecordTree[content=..., left=RecordTree[content=..., right=RecordTree[content=Ikatere...]]
+
+.. _fundamentals-records-get-codec:
+
+Retrieve the Record Codec
+-------------------------
+
+You can use the ``RecordCodecProvider`` to retrieve the
+record codec. You should use this codec to encode and decode Java record objects
+to and from corresponding BSON types while minimizing duplicate code.
```

Comment:
Q: how does using the RecordCodecProvider minimize duplicate code?

Reply:
This is inherent to the codec, so I feel it would be repetitive to add here. It is basically that you can just use the codec instead of needing to create BSON converters for each of your objects.